nabavka/src/main/java/raf/si/racunovodstvo/nabavka/converters/impl/ArtikalConverter.java

```diff
@@ -0,0 +1,48 @@
+package raf.si.racunovodstvo.nabavka.converters.impl;
+
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Component;
+import raf.si.racunovodstvo.nabavka.converters.IConverter;
+import raf.si.racunovodstvo.nabavka.model.Artikal;
+import raf.si.racunovodstvo.nabavka.model.KalkulacijaArtikal;
+import raf.si.racunovodstvo.nabavka.model.KonverzijaArtikal;
+import raf.si.racunovodstvo.nabavka.requests.ArtikalRequest;
+
+@Component
+public class ArtikalConverter implements IConverter<ArtikalRequest, Artikal> {
+
+    private final ModelMapper modelMapper;
+
+    public ArtikalConverter(ModelMapper modelMapper) {
+        this.modelMapper = modelMapper;
+    }
+
+    @Override
+    public Artikal convert(ArtikalRequest source) {
+        Artikal converted;
+        if (source.isAktivanZaProdaju()) {
+            KalkulacijaArtikal mapped = modelMapper.map(source, KalkulacijaArtikal.class);
+            calculateCommonFields(mapped);
+            calculateKalkulacijaFields(mapped);
+            converted = mapped;
+        } else {
+            converted = modelMapper.map(source, KonverzijaArtikal.class);
+            calculateCommonFields(converted);
+        }
+        return converted;
+    }
+
+    private void calculateKalkulacijaFields(KalkulacijaArtikal artikal) {
```

Comment:
Ako sam dobro razumeo iz specifikacije, moze se poslati **prodajnaCena** i onda treba izvoditi ostale atribute. Takodje se javlja problem ako se proslede marza i prodajnaCena u jednom requestu sta je "jace". Marza zavisi od prodajneCene i prodajnaCena zavisi od marze.

Reply:
U specifikaciji pise da se promene desavaju "live". Ja sam to razumeo kao da treba da se racunaju odmah na frontu. To bi trebalo da se proveri sa nekim iz tima za specifikaciju, da ne bismo dva puta radili istu stvar.

Reply:
Moze to na frontu da se racuna, da li onda treba da handlujemo ako neko napravi curl zahtev?
Evo ga ovaj [task na frontu](https://github.com/RAF-SI-2021/Racunovodstvo-Front/issues/47), ali ako sam dobro shvatio iz njihovog zahteva, moguce je uneti samo marzu, a ne prodajnuCenu, mozda treba im to napomenuti. Da li mozes da se cujes sa Milenom da znamo da ce oni da racunaju na frontu?